odl/tomo/backends/stir_bindings.py

```diff
@@ -0,0 +1,293 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+
+"""Backend for STIR: Software for Tomographic Reconstruction
+
+Back and forward projectors for PET.
+
+`ForwardProjectorByBinWrapper` and `BackProjectorByBinWrapper` are general
+objects of STIR projectors and back-projectors, these can be used to wrap a
+given projector.
+
+`projector_from_file` allows users a easy way to create a
+`ForwardProjectorByBinWrapper` by giving file paths to the required templates.
+
+See the STIR `webpage
+<http://stir.sourceforge.net/>`_ for more information.
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+from odl.operator.operator import Operator
+from odl.discr.lp_discr import uniform_discr
+from odl.space.fspace import FunctionSpace
+from odl.set.domain import Cuboid
+
+try:
+    import stir
+    STIR_AVAILABLE = True
+except ImportError:
+    STIR_AVAILABLE = False
+
+__all__ = ('ForwardProjectorByBinWrapper',
+           'BackProjectorByBinWrapper',
+           'stir_projector_from_file',
+           'STIR_AVAILABLE')
+
+
+class StirVerbosity(object):
+    """ Set STIR verbosity to a fixed level """
+    def __init__(self, verbosity):
+        self.verbosity = verbosity
+
+    def __enter__(self):
+        self.old_verbosity = stir.Verbosity.get()
+        stir.Verbosity.set(self.verbosity)
+
+    def __exit__(self, *_):
+        stir.Verbosity.set(self.old_verbosity)
+
+
+class ForwardProjectorByBinWrapper(Operator):
+
+    """ A forward projector using STIR.
+
+    Uses "ForwardProjectorByBinUsingProjMatrixByBin" as a projector.
+    """
+
+    def __init__(self, dom, ran, volume, proj_data,
+                 projector=None, adjoint=None):
+        """ Initialize a new projector.
+
+        Parameters
+        ----------
+        dom : `DiscreteLp`
+            Volume of the projection. Needs to have the same shape as
+            ``volume.shape()``.
+        ran : `DiscreteLp`
+            Projection space. Needs to have the same shape as
+            ``proj_data.to_array().shape()``.
+        volume : `stir.FloatVoxelsOnCartesianGrid`
```

Comment:
Is this an actual array with grid info in memory or just an empty template to set up the grid? `volume` suggests a proper data structure (same with `proj_data`).